### ISSUE TITLE
[OpenBMC] Implement a new command to control OpenPOWER P9 behavior when PSU is cycled

### ISSUE THREAD

**Issue by whowutwut** [xcat2/xcat-core#3724]
I found out that the OpenBMC firmware, in later versions, will remember the last state of the Host power when the power is physically cycled (power cable pulled) .  So if the host was booted before power was lost, and then applied again, it will boot up.  

This differs from the older OpenPOWER servers behavior where if power is pulled and re-applied, the BMC will boot up ONLY, but the host will be powered off until we run `rpower <> on/boot`  

From a conversation... 

OpenBMC Dev:
> Newer code remembers the last requested state, and after a PDU cycle, if the last requested state was a power on, it will auto power on.  so be sure to just power off before doing bmc reboots if you don’t want to auto power back on.  There’s also a setting you can set to disable that

Me: 
> Thanks will look into this..  We would have to disable this and continue to disable on each update of the FW?

OpenBMC Dev:
> it will be a persistent setting, so will be saved accross firmware updates.  With the new BMC code update coming, it may be reset with updates until we get that in.

So,  need to keep this item open and find out the flag and provide a new external that allows the user to be able to switch this behavior on or off.. 

Should target the 2.13.7 release...

**Comment by whowutwut** [comment]
Here's the documentation for the OpenBMC API: [RestorePolicy](https://github.com/openbmc/phosphor-dbus-interfaces/blob/master/xyz/openbmc_project/Control/Power/RestorePolicy.interface.yaml)

**Comment by whowutwut** [comment]
I've opened issue https://github.com/openbmc/openbmc/issues/2186 to change back the policy to default AlwaysOff.  But we should come up with this function support....  

External Design needed for this to start discussion and move it forward.   It's lower priority, but if there are cycles, we can implement it now.

**Comment by gurevichmark** [comment]
To query:

```
[root@briggs01 ~]#curl -k -b cjar -X GET -H "Content-Type: application/json" https://172.11.139.13/xyz/openbmc_project/control/host0/power_restore_policy
{
  "data": {
    "PowerRestorePolicy": "xyz.openbmc_project.Control.Power.RestorePolicy.Policy.AlwaysOn"
  },
  "message": "200 OK",
  "status": "ok"
}
```

To set:
```
[root@briggs01 ~]#  curl -k -b cjar -XPUT -H "Content-Type: application/json" -d '{"data":"xyz.openbmc_project.Control.Power.RestorePolicy.Policy.AlwaysOn"}' https://172.11.139.13/xyz/openbmc_project/control/host0/power_restore_policy/attr/PowerRestorePolicy
{
  "data": null,
  "message": "200 OK",
  "status": "ok"
}
```

Possible values:
```
xyz.openbmc_project.Control.Power.RestorePolicy.Policy.Restore
xyz.openbmc_project.Control.Power.RestorePolicy.Policy.AlwaysOn
xyz.openbmc_project.Control.Power.RestorePolicy.Policy.AlwaysOff
```

**Comment by zet809** [comment]
Let's close this task since the interface `rspconfig powerrestorepolicy` is implemented in #4634.
